extensions/openai-translator/src/views/content.tsx

```diff
@@ -0,0 +1,267 @@
+            if (ref.current) {
+              const newText =
+                ["”", '"', "」"].indexOf(ref.current[ref.current.length - 1]) >= 0 // FIXME 避免显性引用
+                  ? ref.current.slice(0, -1)
+                  : ref.current;
+              setTranslatedText(newText);
+              const record: Record = {
+                id: uuidv4(),
+                mode,
+                created_at: new Date().toISOString(),
+                result: {
+                  from: detectFrom,
+                  to: detectTo,
+                  original: text,
+                  text: newText,
+                },
+              };
+              history.add(record);
+            }
+            query.updateQuerying(false);
+          }
+        },
+        onError: (error) => {
+          onTranslationError(toast, detectFrom, "Error", error);
+        },
+      },
+      id: "querying",
+    };
+    setTranslatedText("");
+    setQuerying(_querying);
+    query.updateText("");
+    translate(_querying.query, entrypoint, apikey);
+  }
+
+  useEffect(() => {
+    if (query.querying && !querying) {
+      doQuery();
+    } else if (!query.querying) {
+      if (querying) {
+        setQuerying(null);
+      }
+    }
+  }, [query.querying]);
+
+  useEffect(() => {
+    updateData();
+  }, [history.data, querying]);
+
+  useEffect(() => {
+    ref.current = translatedText;
+  }, [translatedText]);
+
+  const getQueryingActionPanel = () => (
+    <ActionPanel>
+      <ActionPanel.Submenu title="Cancel">
+        <Action
+          title="Abort"
+          icon={Icon.Stop}
+          shortcut={{ modifiers: ["ctrl"], key: "c" }}
+          onAction={() => {
+            if (querying) {
+              querying.controller.abort();
+            }
+          }}
+        />
+      </ActionPanel.Submenu>
+    </ActionPanel>
+  );
+
+  const getRecordActionPanel = (record: Record) => (
+    <ActionPanel>
+      {query.text && <Action title={capitalize(mode)} icon={Icon.Book} onAction={() => query.updateQuerying(true)} />}
+      <Action
+        title={`Switch to Translate ${query.langType == "To" ? "From" : "To"}`}
+        icon={Icon.Switch}
+        onAction={() => {
+          query.updateLangType(query.langType == "To" ? "From" : "To");
+        }}
+      />
+      <ActionPanel.Section title="Copy">
+        <Action.CopyToClipboard
+          title="Copy Translation"
+          content={record.result.text ?? ""}
+          shortcut={{ modifiers: ["cmd", "ctrl"], key: "c" }}
+        />
+        <Action.CopyToClipboard
+          title="Copy Original"
+          content={record.result.original ?? ""}
+          shortcut={{ modifiers: ["cmd", "shift"], key: "c" }}
+        />
+      </ActionPanel.Section>
+      {getLoadActionSection(record, (str) => {
+        query.updateText(str);
+      })}
+      {getModeActionSection((mode) => {
+        setMode(mode);
+      })}
+      <Action
+        title="Delete Translation"
+        icon={Icon.Trash}
```

Comment:
```suggestion
        icon={Icon.Trash}
        style={Action.Style.Destructive}
```

Reply:
Maybe even add a [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) if it makes sense?

Reply:
Yes!